src/core/document.js

```diff
@@ -1711,13 +1711,22 @@ class PDFDocument {
       : clearGlobalCaches();
   }
 
-  #collectFieldObjects(name, fieldRef, promises, annotationGlobals) {
-    const field = this.xref.fetchIfRef(fieldRef);
+  async #collectFieldObjects(name, fieldRef, promises, annotationGlobals) {
+    const { xref } = this;
+
+    if (!(fieldRef instanceof Ref)) {
+      return;
+    }
+    const field = await xref.fetchAsync(fieldRef);
     if (!(field instanceof Dict)) {
       return;
     }
     if (field.has("T")) {
-      const partName = stringToPDFString(field.get("T"));
+      let title = field.getRaw("T");
```

Comment:
I think we should have a method `field.getAsync` and then use something like `const partName = stringToPDFString(await field.getAsync("T"));`

Reply:
As mentioned at the end of the commit message, the idea was to limit the amount of asynchronicity a bit here since these are usually direct objects.

Reply:
I created a pdf with a lot of fields (1121):
[lot_of_fields.pdf](https://github.com/mozilla/pdf.js/files/13297078/lot_of_fields.pdf)
and in using the Firefox profiler I don't see any real difference with/without your patch:
 - without: https://share.firefox.dev/3SrJIlr
 - with: https://share.firefox.dev/40wNqwg

So I suppose that it won't make any difference overall.

I know that adding some asynchronicity can slightly increase the overhead, but for a "normal" pdf it's probably almost nothing and for some pdf with a large and complex structure it could be slightly worst, but the real bottleneck would be related to the structure of the pdf itself and not really because of asynchronicity, that said it's a feeling/opinion...

Reply:
Thank you; I've simplified this to use `getAsync` directly (with one extra `has(...)` check added since most fields don't have any kids).

Reply:
Here's a profile with the updated patch:
https://share.firefox.dev/3QMV4PJ
The total overhead is observable (~4ms) but in general for "normal" pdfs it should be invisible.